dft util: add specific unit tests

tcoratger opened the pull request.

SyxtonPrime reviewed (approved):
These look fine but, tbh, all of these are tested through `dft_testing` as they are called in a lot of the `dft` code so it seems less important to have specific tests here.

We can put a few in if you think it's worth it but I'd leave aggressively towards only a couple.

tcoratger commented on dft/src/util.rs:
```diff
@@ -34,3 +34,200 @@ pub(crate) fn coset_shift_cols<F: Field>(mat: &mut RowMajorMatrix<F>, shift: F)
+        ];
+
+        assert_eq!(mat.values, expected);
+    }
```
no problem just removed

tcoratger commented on dft/src/util.rs:
```diff
@@ -34,3 +34,200 @@ pub(crate) fn coset_shift_cols<F: Field>(mat: &mut RowMajorMatrix<F>, shift: F)
+        ];
+
+        assert_eq!(mat.values, expected);
+    }
```
removed

tcoratger commented:
> These look fine but, tbh, all of these are tested through `dft_testing` as they are called in a lot of the `dft` code so it seems less important to have specific tests here.
> 
> We can put a few in if you think it's worth it but I'd leave aggressively towards only a couple.

No worries, I've deleted the tests you indicated. I just added these additional tests to spot potential bugs upstream on the smallest functions, and to learn how they work :) 

I also deleted the tests on extension fields, but I added them because the functions in question involve division using the prime subfield, so I wanted to make sure it was correct even with an extension field. Let me know if you see a particular use for them, and I can add them back if necessary.

SyxtonPrime commented:
Makes sense, I think the ones that are left look good.
